pkg/driver/rdmadevice.go

```diff
@@ -21,12 +21,39 @@ import (
 
 	"github.com/vishvananda/netlink"
 	"github.com/vishvananda/netns"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// rdmaNetnsModeShared and rdmaNetnsModeExclusive define the RDMA subsystem
+	// network namespace mode. An RDMA device can only be assigned to a network
+	// namespace when the RDMA subsystem is set to an "exclusive" network
+	// namespace mode. When the subsystem is set to "shared" mode, an attempt to
+	// assign an RDMA device to a network namespace will result in failure.
+	// Additionally, "If there are active network namespaces and if one or more
+	// RDMA devices exist, changing mode from shared to exclusive returns error
+	// code EBUSY."
+	//
+	// Ref. https://man7.org/linux/man-pages/man8/rdma-system.8.html
+	rdmaNetnsModeShared    = "shared"
+	rdmaNetnsModeExclusive = "exclusive"
 )
 
 // Based on existing RDMA CNI plugin
 // https://github.com/k8snetworkplumbingwg/rdma-cni
 
 func nsAttachRdmadev(hostIfName string, containerNsPAth string) error {
+	rdmaNetnsMode, err := netlink.RdmaSystemGetNetnsMode()
+	if err != nil {
+		return fmt.Errorf("failed to determine the RDMA subsystem's network namespace mode: %w", err)
+	}
+	if rdmaNetnsMode == rdmaNetnsModeShared {
+		// TODO: Potentially move this check to the function invokers side for
+		// improved visiblity and expectation of what this function does.
+		klog.Info("Skipping setting network namespace for RDMA device since RDMA sybsystem is currently configured for shared mode.")
```

Comment:
I think you can configure it in some way that can be shared, IIRC it creates another type of devices https://github.com/Mellanox/k8s-rdma-shared-dev-plugin/blob/master/pkg/resources/resources_manager.go

Reply:
https://docs.nvidia.com/networking/display/ofedv502180/kubernetes+with+shared+hca

Reply:
see here, it uses macvlan or ipvlan https://spidernet-io.github.io/spiderpool/v0.9/usage/rdma-roce/#introduction

Reply:
That's correct. 

If I'm understanding it correctly, if the subsystem is configured for "shared" mode, we don't have to take any action item with regards to moving the device to the correct namespace (or rather, if it's in "shared" mode, we CANNOT move the device to a network namespace). We just have to ensure that the correct device files (like the character device files in `/dev/infiniband/` are correctly mounted to the container, which this PR doesn't attempt to tackle.)

So basically trying to skip doing anything here if it's in "shared" mode right now. Does I make sense?

Reply:
yes, I just wanted to be sure my understanding is correct.
I will create an issue to follow up on this, I really like to handle all possible scenarios from dranet